File: cdk-lambda-chaos/lib/serverless-api/python/lambda-functions/deleteItem.py
import os
import json
import boto3

dynamodb = boto3.resource('dynamodb')

def handler(event, context):
    # Retrieve table name and primary key from environment variables
    table_name = os.environ.get('TABLE_NAME')
    primary_key = os.environ.get('PRIMARY_KEY')
    
    if not table_name or not primary_key:
        raise ValueError('TABLE_NAME and PRIMARY_KEY environment variables must be set')
    
    table = dynamodb.Table(table_name)
    
    # Assume the resource path parameter is in the 'pathParameters' key of the event
    try:
        resource_id = event['pathParameters']['id']
    except KeyError:
        return {
            'statusCode': 400,
            'body': json.dumps({'error': 'Missing resource ID'})
        }
    
    # Delete the item from DynamoDB
    try:
        response = table.delete_item(Key={primary_key: resource_id})
    except Exception as e:
        return {
            'statusCode': 500,
            'body': json.dumps({'error': str(e)})
        }
    
    if response['ResponseMetadata']['HTTPStatusCode'] == 200:
        return {
            'statusCode': 200,
            'body': json.dumps({'message': 'Item deleted successfully'})
        }
    else:
        return {
            'statusCode': 404,
            'body': json.dumps({'error': 'Item not found'})
        }

File: cdk-lambda-chaos/lib/serverless-api/typescript/lambda-functions/deleteItem.mjs
import { DynamoDBClient } from "@aws-sdk/client-dynamodb";
import {
  DynamoDBDocumentClient,
  ScanCommand,
  PutCommand,
  GetCommand,
  DeleteCommand,
} from "@aws-sdk/lib-dynamodb";

const client = new DynamoDBClient({});
const dynamo = DynamoDBDocumentClient.from(client);

const TABLE_NAME = process.env.TABLE_NAME || 'Orders';
const PRIMARY_KEY = process.env.PRIMARY_KEY || 'Id';

export const handler = async (event, context) => {
    const requestedItemId = event.pathParameters.id;
    if (!requestedItemId) {
      return { statusCode: 400, body: `Error: You are missing the path parameter id` };
    }
    try {
      const data = await dynamo.send(
        new DeleteCommand({
          TableName: TABLE_NAME,
          Key: {
            [PRIMARY_KEY]: requestedItemId
          }
        })
      );
      return { statusCode: 200, body: JSON.stringify(data) };
    } catch (err) {
      console.error(err);
      return { statusCode: 500, body: JSON.stringify(err) };
    }
  
}